src/Menu.cpp

```diff
@@ -537,6 +307,271 @@ void Menu::DrawSettings()
+				"This is activated if the startup compilation is skipped. "
+				"The more threads the faster compilation will finish but may make the system unresponsive. ");
+		}
+
+		if (ImGui::SliderInt("Test Interval", reinterpret_cast<int*>(&testInterval), 0, 10)) {
+			if (testInterval == 0) {
+				inTestMode = false;
+				logger::info("Disabling test mode.");
+				State::GetSingleton()->Load(State::ConfigMode::TEST);  // restore last settings before entering test mode
+			} else if (!inTestMode) {
+				logger::info("Saving current settings for test mode and starting test with interval {}.", testInterval);
+				State::GetSingleton()->Save(State::ConfigMode::TEST);
+				inTestMode = true;
+			} else {
+				logger::info("Setting new interval {}.", testInterval);
+			}
+		}
+		if (auto _tt = Util::HoverTooltipWrapper()) {
+			ImGui::Text(
+				"Sets number of seconds before toggling between default USER and TEST config. "
+				"0 disables. Non-zero will enable testing mode. "
+				"Enabling will save current settings as TEST config. "
+				"This has no impact if no settings are changed. ");
+		}
+		bool useFileWatcher = shaderCache.UseFileWatcher();
+		ImGui::TableNextColumn();
+		if (ImGui::Checkbox("Enable File Watcher", &useFileWatcher)) {
+			shaderCache.SetFileWatcher(useFileWatcher);
+		}
+		if (auto _tt = Util::HoverTooltipWrapper()) {
+			ImGui::Text(
+				"Automatically recompile shaders on file change. "
+				"Intended for developing.");
+		}
+
+		if (ImGui::Button("Dump Ini Settings", { -1, 0 })) {
+			Util::DumpSettingsOptions();
+		}
+		if (!shaderCache.blockedKey.empty()) {
+			auto blockingButtonString = std::format("Stop Blocking {} Shaders", shaderCache.blockedIDs.size());
+			if (ImGui::Button(blockingButtonString.c_str(), { -1, 0 })) {
+				shaderCache.DisableShaderBlocking();
+			}
+			if (auto _tt = Util::HoverTooltipWrapper()) {
+				ImGui::Text(
+					"Stop blocking Community Shaders shader. "
+					"Blocking is helpful when debugging shader errors in game to determine which shader has issues. "
+					"Blocking is enabled if in developer mode and pressing PAGEUP and PAGEDOWN. "
+					"Specific shader will be printed to logfile. ");
+			}
+		}
+		if (ImGui::TreeNodeEx("Addresses")) {
+			auto Renderer = RE::BSGraphics::Renderer::GetSingleton();
+			auto BSShaderAccumulator = RE::BSGraphics::BSShaderAccumulator::GetCurrentAccumulator();
+			auto RendererShadowState = RE::BSGraphics::RendererShadowState::GetSingleton();
+			ADDRESS_NODE(Renderer)
+			ADDRESS_NODE(BSShaderAccumulator)
+			ADDRESS_NODE(RendererShadowState)
+			ImGui::TreePop();
+		}
+		if (ImGui::TreeNodeEx("Statistics", ImGuiTreeNodeFlags_DefaultOpen)) {
+			ImGui::Text(std::format("Shader Compiler : {}", shaderCache.GetShaderStatsString()).c_str());
+			ImGui::TreePop();
+		}
+		ImGui::Checkbox("Extended Frame Annotations", &State::GetSingleton()->extendedFrameAnnotations);
+		TruePBR::GetSingleton()->DrawSettings();
```

Comment:
meant to draw truepbr under addresses? Maybe it's own menuitem?

Reply:
I intend to keep how menus were categorized before, and leave further ui changes to the next commit, but I guess PBR could be separate.